compiler/quilt/test/test_import.py

```diff
@@ -198,6 +198,130 @@ def test_multiple_package_dirs(self):
             from quilt.data.foo import multiple1
             assert multiple1.dataframes
 
+             # Check search order again just to make sure it doesn't return same named empty package
+            from quilt.data.foo import multiple1
+            assert multiple1.dataframes
+
+            teststore = PackageStore(self._store_dir)
+            contents = open(os.path.join(teststore.package_path(None, 'foo', 'multiple1'),
+                                          Package.CONTENTS_DIR,
+                                          multiple1._package.get_hash())).read()
+            assert contents
+
+            # bad imports
+            with self.assertRaises(ImportError):
+                from quilt.data.nonexisting import multiple2
+            with self.assertRaises(ImportError):
+                from quilt.data.foo import nonexistingdata
+            with self.assertRaises(ImportError):
+                import quilt.data.foo.nonexisting
+            with self.assertRaises(ImportError):
+                import quilt.data.bad_user.nonexisting
+            with self.assertRaises(ImportError):
+                from quilt.data.foo.dataframes import blah
+            with self.assertRaises(ImportError):
+                from quilt.data.foo.baz import blah
+
+        # add same dir to the search path
+        dirs = 'foo/%s:%s:%s' % (PACKAGE_DIR_NAME, new_build_dir, new_build_dir)
+        with patch.dict(os.environ, {'QUILT_PACKAGE_DIRS': dirs}):
+            from quilt.data.foo import multiple2
+
+        # check that invalid folder names fail
+        bad_build_dir = '?\*;:|<>'
+        with patch.dict(os.environ, {'QUILT_PRIMARY_PACKAGE_DIR': bad_build_dir}):
+            with self.assertRaises(AssertionError):
```

Comment:
This is basically testing a misuse of the API. We may consider doing something different if the environment variable is bad - we shouldn't really test assertion errors

Reply:
@dimaryaz is it not a legit concern that someone would provide a malformed build dir?  and if so, we get the same error (and not e.g. a stack trace)

Reply:
Sure. In that case, we should change the code to not raise an `AssertionError` and add a test for the new behavior.